### PR TITLE
Add daily completed mission report and solution for storing missions completed in the past.

### LINKED ISSUE #41: Add new day report
After sleeping, check if any missions were completed, and show a report.

The report should be an automated message in the messages tab

It should be a separate UI that looks like a browser window to an internal website

Handle the outcome of missions

### PR DESCRIPTION
**Pull request by maxwellpark**
The past mission data can later be used in the Analytics system.

### CODE REVIEW

**Comment by maxwellpark** [comment]
This is somewhat of a WIP. But I have tested it and the core functionality works.
I need your opinion on the mission archiving. 
I moved some files around to make them more logically placed.

File: Assets/Scripts/DataModels/Missions/MissionOutcomes/PilotXpOutcome.cs [hunk 1]
```diff
@@ -8,11 +8,18 @@ public class PilotXpOutcome : MissionOutcome
 	
 	public override void Process(Mission mission) 
 	{
-		PilotsManager.Instance.AwardXp(
-            mission.Ship, Mathf.FloorToInt(Random.Range(xpMin, xpMax) * ApplyOmens(mission)));
+        int xpGained = Mathf.FloorToInt(Random.Range(xpMin, xpMax) * ApplyOmens(mission));
+        PilotsManager.Instance.AwardXp(mission.Ship, xpGained);
```
Comment (ambid17) [inline Assets/Scripts/DataModels/Missions/MissionOutcomes/PilotXpOutcome.cs]: pilots manager shouldn't need the ship

File: Assets/Scripts/DataModels/Ships/Ship.cs [hunk 2]
```diff
@@ -35,6 +35,14 @@ public float GetHullPercent()
     public float GetFuelPercent()
     {
         return (float)saveData.currentFuel / maxFuel;
+    }
+
+    public void DeductFuel()
+    {
+        if (CurrentMission != null)
```
Comment (ambid17) [inline Assets/Scripts/DataModels/Ships/Ship.cs:42]: add "else" condition with debug

File: Assets/Scripts/Singletons/DataModelSingletons/ArchivedMissionsManager.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+public class ArchivedMissionsManager : MonoBehaviour, IDataModelManager
+{
+    public static ArchivedMissionsManager Instance { get; private set; }
+    public List<ArchivedMission> ArchivedMissions { get; set; }
+
+    /// <summary>
+    /// Missions that were completed yesterday.
+    /// These are ready to be displayed in the daily mission report.
+    /// </summary>
+    public List<ArchivedMission> MissionsCompletedYesterday { get; set; }
```
Comment (ambid17) [inline Assets/Scripts/Singletons/DataModelSingletons/ArchivedMissionsManager.cs:14]: clearing this should happen in this file

File: Assets/Scripts/Singletons/DataModelSingletons/DataModelManagerInitialiser.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+{
+    private void Start()
+    {
+        InitDataModelManagers();
+    }
+
+    private void InitDataModelManagers()
+    {
+        // MissionsManager must exist before ArchivedMissionsManager.
+        MissionsManager.Instance.Init();
```
Comment (ambid17) [inline Assets/Scripts/Singletons/DataModelSingletons/DataModelManagerInitialiser.cs:13]: need to decouple these as this is gonna need to be async in the future

File: Assets/Scripts/Singletons/DataModelSingletons/PlayerManager.cs [hunk 5]
```diff
@@ -57,6 +48,11 @@ public void Init()
             DataModelsUtils.CreateSaveFolder(PlayerData.FOLDER_NAME);
         }
         playerMovement = FindObjectOfType<PlayerMovement>();
+
+        if (playerData == null)
```
Comment (ambid17) [inline Assets/Scripts/Singletons/DataModelSingletons/PlayerManager.cs:52]: this check should probably be added everywhere after data loading

File: Assets/Scripts/UI/Terminal/NewDayReportUI/NewDayReportUI.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        get => ArchivedMissionsManager.Instance.MissionsCompletedYesterday[currentReportIndex];
+    }
+
+    private void Start() => BedCanvasUI.OnEndOfDay += () => HasBeenViewed = false;
+    private void OnDisable() => HasBeenViewed = true;
+
+    public void Init()
+    {
+        reportCardInstance = Instantiate(reportCardPrefab, transform);
```
Comment (ambid17) [inline Assets/Scripts/UI/Terminal/NewDayReportUI/NewDayReportUI.cs]: dont instantiate every time. just turn it on/off and change the text

**Review by ambid17 — CHANGES_REQUESTED** [review]
for the love of god. fix line endings

**Review by ambid17 — APPROVED** [review]

**Pull request merged**